packages/dd-trace/src/appsec/standalone.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const { channel } = require('dc-polyfill')
+const startCh = channel('dd-trace:span:start')
+const { APM_TRACING_ENABLED_KEY, APPSEC_PROPAGATION_KEY } = require('../constants')
+
+let enabled
+
+function isStandaloneEnabled () {
+  return enabled
+}
+
+function onSpanStart ({ span, fields }) {
+  const { parent } = fields
+  const tags = span.context()._tags
+
+  if (!parent || parent._isRemote) {
+    tags[APM_TRACING_ENABLED_KEY] = 0
+  }
+}
+
+function configure (config) {
+  enabled = config.appsec?.standalone?.enabled
+
+  if (enabled) {
+    startCh.subscribe(onSpanStart)
+  } else {
+    startCh.unsubscribe(onSpanStart)
+  }
+}
+
+function sample (span) {
+  if (enabled) {
+    span.context()._trace.tags[APPSEC_PROPAGATION_KEY] = '1'
```

Comment:
why are we not using setTag ?

Reply:
As it is a propagation tag (`_dd.p.appsec`) it's only propagated if it is added to the trace tags.
`setTag` includes the tag in the span tags